packages/headless/src/features/facets/facet-set/facet-set-slice.ts

```diff
@@ -189,27 +189,29 @@ export const facetSetReducer = createReducer(
         }
 
         const searchResultValue = buildSelectedFacetValueRequest(rawValue);
-        const firstIdleIndex = currentValues.findIndex(
-          (v) => v.state === 'idle'
-        );
-        const indexToInsertAt =
-          firstIdleIndex === -1 ? currentValues.length : firstIdleIndex;
-
-        const valuesBefore = currentValues.slice(0, indexToInsertAt);
-        const valuesAfter = currentValues.slice(indexToInsertAt + 1);
-
-        facetRequest.currentValues = [
-          ...valuesBefore,
-          searchResultValue,
-          ...valuesAfter,
-        ];
-        facetRequest.numberOfValues = facetRequest.currentValues.length;
+        insertValueBeforeIdleValues(facetRequest, searchResultValue);
         facetRequest.freezeCurrentValues = true;
         facetRequest.preventAutoSelect = true;
       });
   }
 );
 
+function insertValueBeforeIdleValues(
+  facetRequest: FacetRequest,
+  facetValue: FacetValueRequest
+) {
+  const {currentValues} = facetRequest;
+  const firstIdleIndex = currentValues.findIndex((v) => v.state === 'idle');
```

Comment:
There might be one flaw here if the `selected` are not all grouped together at the beginning of the array, and there's gap where an idle value is inserted before.

Normally it "should not" happen, but... 
What if instead we used the last index for `selected` state ? So `insertValueBeforeIdleValues` -> `insertValueAfterSelectedValues`

Reply:
So before
```
[✓] Selected
[✓] New
[.  ] Idle
[✓] Selected
[.  ] Idle
```

with your proposition

```
[✓] Selected
[.  ] Idle
[✓] Selected
[✓] New
[.  ] Idle
```

is that it?

Reply:
Yeah I mean it's still weird anyway 🤷

Reply:
Hmm, the logic should insert correctly when the selected values are not grouped at the beginning. It's possible to test out the UX by selecting a facet search result with different facet values pre-selected.

Note that it _replaces_ the first idle value, not insert before the idle value.